core/lib/dal/src/transactions_dal.rs

```diff
@@ -1752,6 +1752,36 @@ impl TransactionsDal<'_, '_> {
         Ok(rows.len())
     }
 
+    /// Resets `in_mempool` to `FALSE` for the given transaction hashes.
+    pub async fn reset_mempool_status(&mut self, transaction_hashes: &[H256]) -> DalResult<()> {
+        // Convert H256 hashes into `&[u8]`
+        let hashes: Vec<_> = transaction_hashes.iter().map(H256::as_bytes).collect();
+
+        // Execute the UPDATE query
+        let result = sqlx::query!(
+            r#"
+            UPDATE transactions
+            SET
+                in_mempool = FALSE
+            WHERE
+                hash = ANY($1)
+            "#,
+            &hashes as &[&[u8]]
+        )
+        .instrument("return_to_mempool#update_to_false")
+        .with_arg("transaction_hashes.len", &hashes.len())
+        .execute(self.storage)
+        .await?;
+
+        // Log debug information about how many rows were affected
```

Comment:
Bikeshedding: IMO, it'd make more sense to return the number of affected rows from the method and do logging on the caller side. This could e.g. make logging targets more consistent.

Reply:
Looking at the file it seems like there are a lof of places where `rows_affected` is printed in place, so I guess it makes more sense to do it the same way here